src/services/dexes/uniswapV3/UniswapV3LiquidityResolver.ts

```diff
@@ -0,0 +1,159 @@
+    ]);
+
+    if (!tokenA) {
+      throw new Error(`${this.logPrefix}[${chainId}] token0: ${token0} not found`);
+    }
+
+    if (!tokenB) {
+      throw new Error(`${this.logPrefix}[${chainId}] token1: ${token1} not found`);
+    }
+
+    const UniswapV3ChainNumber = this.getUniswapV3ChainNumber(chainId);
+
+    if (!UniswapV3ChainNumber) {
+      throw new Error(`${this.logPrefix}[${chainId}] chain number found`);
+    }
+
+    return [
+      new Token(UniswapV3ChainNumber, tokenA.address, tokenA.decimals, tokenA.symbol, tokenA.name),
+      new Token(UniswapV3ChainNumber, tokenB.address, tokenB.decimals, tokenB.symbol, tokenB.name),
+    ];
+  }
+
+  private async processParams(uniswapV3Params: UniswapV3Param[], chainId: ChainsIds) {
+    await Promise.all(
+      uniswapV3Params
+        .filter((param) => param.fromChain.includes(chainId))
+        .map(async (param) => {
+          const pools = await this.uniswapV3PoolRepository.find({
+            token0: param.token0,
+            token1: param.token1,
+            fromChain: [chainId],
+            protocol: this.protocol,
+          });
+
+          if (pools.length === 0) {
+            this.logger.error(`${this.logPrefix}[${chainId}] pool not found for token ${param.token0}-${param.token1}`);
+            return;
+          }
+
+          const [token0, token1] = await this.getTokens(param.token0, param.token1, chainId);
+
+          await Promise.all(
+            pools.map(async (pool) => {
+              const liquidity = await this.getLiquidities(pool, token0, token1, chainId);
+              if (liquidity) await this.saveLiquidities(pool, token0, token1, chainId, liquidity);
+            }),
+          );
+        }),
+    );
+  }
+
+  private async getLiquidities(pool: UniswapV3Pool, token0: Token, token1: Token, chainId: ChainsIds) {
+    const liquidity = await this.liquiditySummingService.apply(pool.address, token0, token1, pool.fee, chainId);
+    return liquidity;
+  }
+
+  private async saveLiquidities(
+    pool: UniswapV3Pool,
+    token0: Token,
+    token1: Token,
+    chainId: ChainsIds,
+    liquidity: {
+      liquidityActive: string;
+      liquidityLockedToken0: number;
+      liquidityLockedToken1: number;
+    },
+  ) {
+    const save = await this.uniswapV3PoolRepository.saveLiquidity(
+      {
+        address: pool.address,
+        chainId,
+        token0: token0.address,
+        token1: token1.address,
+        fee: pool.fee,
+      },
+      liquidity,
+    );
+
+    if (!save) {
+      this.logger.error(`${this.logPrefix}[${chainId}] Update liquidity failed`);
+    }
+
+    if (save) {
+      this.logger.info(
+        // eslint-disable-next-line max-len
+        `${this.logPrefix}[${chainId}] data saved for pool address: ${pool.address}, token0: ${token0.address}, token1: ${token1.address}`,
+      );
+    }
+  }
+
+  private getUniswapV3ChainNumber(chainId: ChainsIds) {
+    const uniswapV3ChainNumbers: Partial<Record<ChainsIds, number>> = {
```

Comment:
I think this code is duplicated, let's put it in one place

Reply:
The const was outdated, we only use this get method